source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1187,6 +1297,25 @@ void Filter::mergePerRouteConfig() {
   }
 }
 
+void DeferredDeletableStream::closeStreamOnTimer(Filter* filter) {
+  // Close the stream.
+  if (stream_) {
+    ENVOY_LOG(debug, "Closing the stream");
+    stream_->close();
+    stream_.reset();
+  } else {
+    ENVOY_LOG(debug, "Stream already closed");
+  }
+
+  // Erase this entry from the map.
+  parent.erase(filter);
+}
+
+void DeferredDeletableStream::deferredClose(Envoy::Event::Dispatcher& dispatcher, Filter* filter) {
+  derferred_close_timer = dispatcher.createTimer([this, filter] { closeStreamOnTimer(filter); });
```

Comment:
You need independent testing for all of this, outside of integration tests.

Reply:
I used simulate time in integration test to ensure this deferred closure behavior and  timer function are tested.
`timeSystem().advanceTimeWaitImpl(std::chrono::milliseconds(DEFAULT_CLOSE_TIMEOUT_MS));`

Sure, I can look into this to add more unit test.